plutus-metatheory/src/Algorithmic.lagda

```diff
@@ -5,12 +5,12 @@ module Algorithmic where
 ## Imports
 
 \begin{code}
-open import Data.Product using (Σ;proj₁;proj₂;_×_) renaming (_,_ to _,,_)
-open import Data.List using (List;[])
+open import Data.List using (List;[];replicate;_++_)
 open import Relation.Binary.PropositionalEquality using (_≡_;refl)
+open import Data.List.NonEmpty using (length)
 
 open import Utils hiding (TermCon)
-open import Type using (Ctx⋆;∅;_,⋆_;_⊢⋆_;_∋⋆_;Z;S)
+open import Type using (Ctx⋆;∅;_,⋆_;_⊢⋆_;_∋⋆_;Z;S;Φ)
```

Comment:
I kinda find this implicit binding of variables a bit confusing, but I certainly understand the appeal and the convenience.

Reply:
It gets some time to get used, but in the end it removes noise.